docs/example-library/meltano-env-precedence/ending-meltano.yml

```diff
@@ -0,0 +1,39 @@
+version: 1
+project_id: faa14098-8fdc-4336-850e-96db154b7990
+send_anonymous_usage_stats: false
+env:
+  # Level 2: top-level `env:`
+  # Inherits from terminal context
+  LEVEL_NUM: '2'                  #  '2'
+  STACKED: ${STACKED}2            # '12'
+plugins:
+  utilities:
+  - name: example
+    namespace: example
+    # Env vars in `pip_url` are expanded during `meltano install`.
+    # Note that `meltano install` runs with no active environment,
+    # so 3 and 5 are absent from the expanded string.
+    pip_url: -e env-var-in-pip-url-example-${STACKED}  # 'env-var-in-pip-url-example-124'
+    executable: example
+    env:
```

Comment:
This simplifies reproducibility by declaring a couple `echo` commands which print out the respective values.

```suggestion
    pip_url: -e env-var-in-pip-url-example-${STACKED}  # 'env-var-in-pip-url-example-124'
    settings:
    - name: echo_text
      env: ECHO_TEXT
    config:
      echo_text: ${STACKED}
    commands:
      echo-stacked:        
        # Echoed text is 1234 (no environment) or 12345 (in 'prod')
        executable: echo
        args: "Stacked env var value is: ${STACKED}"
      echo-config:
        # FIXME: Should be same as echo_stacked 1234.
        #        Currently prints partially resolved: '${STACKED}4'
        executable: echo
        args: "Config value is: ${ECHO_TEXT}"
    env:
```

Example usage shows env vars are working correctly when referenced as env vars:

```console
cd docs/example-library/meltano-env-precedence
poetry run meltano invoke example:echo-stacked
> Stacked env var value is: 1234
poetry run meltano --environment=prod invoke example:echo-stacked
> Stacked env var value is: 12345
```

However the stacked referenced does not appear to be correctly resolved when the config value references it:

```console
poetry run meltano invoke example:echo-config
> Echoed text is: ${STACKED}4
poetry run meltano --environment=prod invoke example:echo-config
> Echoed text is: ${STACKED}5
```

I hope this is helpful. I authored and tested the above in codespaces. I could put this in a separate PR but it seems that would probably just slow you down. Lmk.

Reply:
Note: In order to invoke this, I had to comment out the `pip_url`. Is there a way to make this 'real' so it is runnable but also demonstrate the use of env vars? Maybe pass the name of a real repo, pypi image, or local path?

Even after commenting out `pip_url`, Meltano will still complain about 'plugin not installed' if it can't find the executable. Easy fix for that is to delete `executable: example` (it's not required) or replace with `executable: echo`.

Reply:
@WillDaSilva - Per your comment here, I understand the scope of this PR is not to resolve precedence/resolution issues broadly. Totally fine to hold off on the above suggestions and keep these for a later PR.

Reply:
A little late, I just saw the related bug report here:

- https://github.com/meltano/meltano/issues/7023

Reply:
> I hope this is helpful.

It was, thanks!

154f1c6cf4a5051d07eea95f52f1f817c4112c30

I was a little surprised that `meltano invoke` cannot invoke executables from outside the plugin venv. I made a minimal Python module that provides an `echo` command to work around that.

Reply:
@aaronsteers I also opened an issue to track the problem you found with `meltano invoke example:echo-config`: 
- https://github.com/meltano/meltano/issues/7034